gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/test/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/SSOSessionHandlerTest.java

```diff
@@ -254,6 +256,52 @@ public void shouldInvoke_differentClient_sameIdp() throws Exception {
                 HttpStatusCode.OK_200, "OK");
     }
 
+
+    @Test
+    public void shouldInvoke_differentClient_sameIdp_using_UserIdentities() throws Exception {
+        User user = new User();
+        user.setId("user-id");
+        user.setClient("test-client");
+        user.setSource("idp-1");
+        UserIdentity userIdentity = new UserIdentity();
+        userIdentity.setProviderId("idp-2");
+        user.setIdentities(List.of(userIdentity));
+
+        Client client = new Client();
+        client.setId("client-id");
+        client.setClientId("test-client");
+
+        Client requestedClient = new Client();
+        requestedClient.setId("client-requested-id");
+        requestedClient.setClientId("requested-client");
+        requestedClient.setIdentityProviders(getApplicationIdentityProviders("idp-2"));
+
+        when(clientSyncService.findById(anyString())).thenReturn(Maybe.empty()).thenReturn(Maybe.empty());
```

Comment:
question: is double thenReturn() a minor mistake or intentionally for some reason?

Reply:
it is a mistake. thanks